===================================================================
RCS file: /cvsroot/mailutils/cvsroot/mailutils/mailutils/mailbox/header.c,v
retrieving revision 1.65
retrieving revision 1.66
diff -u -r1.65 -r1.66
--- mailutils/mailutils/mailbox/header.c	2004/06/02 11:03:36	1.65
+++ mailutils/mailutils/mailbox/header.c	2005/05/13 09:39:57	1.66
@@ -710,7 +710,7 @@
     {
       /* save one for the null */
       --buflen;
-      len = (len > buflen) ? len : len;
+      len = (len > buflen) ? buflen : len;
       memcpy (buf, header->hdr[num].fn, len);
       buf[len] = '\0';
     }
